src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -0,0 +1,308 @@
+#include "cFE_util.h"
+#include "osfilesys.h"
+
+struct FS fs[MAX_NUM_FS];
+struct fsobj files[MAX_NUM_FILES];
+
+/*
+ * Notes on this version:
+ * This is not the final implementation of a file system for the cFE on COS
+ * For now the file system is read only and uses static memory for fsobj and FS
+ * The next step is to write a dumb allocator/deallocator and use it for 
+ * new fs, fsobj, and file data
+ * I have been intentional to restrict memory specific details to this file
+ * osfilesys.h and osfiles.c 'should' not care about how FS, fsobj, and data
+ * are stored in memory.
+ */
+
+/******************************************************************************
+** Tar Level Methods
+******************************************************************************/
+
+/* 
+ * Loads the position in memory of linked
+ * Checks for badly linked or no linked tar file.
+ */
+uint32 tar_load()
+{
+    if(! &_binary_cFEfs_tar_size)                         return OS_FS_ERR_DRIVE_NOT_CREATED;
```

Comment:
to be completely honest I am not 100% sure why &_binary_cFEfs_tar_size gives me the size of the fs but it does.  It feels weird to me and the symbol names produced by the linker are not very useful.  The check makes sure that the size of the tar is greater than 0.  I'm working on commenting this section.